zulip_bots/zulip_bots/bots/googletranslate/test_googletranslate.py

```diff
@@ -0,0 +1,72 @@
+#!/usr/bin/env python
+
+from __future__ import absolute_import
+from __future__ import print_function
+
+import json
+
+from unittest.mock import patch
+from requests.exceptions import HTTPError, ConnectionError
+
+from zulip_bots.test_lib import BotTestCase
+from zulip_bots.bots.googletranslate.googletranslate import TranslateError
+
+help_text = '''
+Google translate bot
+Please format your message like:
+`@-mention "<text_to_translate>" <target-language> <source-language(optional)>`
+Visit [here](https://cloud.google.com/translate/docs/languages) for all languages
+'''
+
+class TestGoogleTranslateBot(BotTestCase):
+    bot_name = "googletranslate"
+
+    def test_normal(self):
+        with self.mock_config_info({'key': 'abcdefg'}), \
+                self.mock_http_conversation('test_normal'):
+                    with self.mock_http_conversation('test_languages'):
+                        self.initialize_bot()
+                    self.assert_bot_response(
+                        message = {'content': '"hello" de', 'sender_full_name': 'tester'},
+                        response = {'content': 'Hallo (from tester)'},
+                        expected_method = 'send_reply'
+                    )
+
+    def test_403(self):
+        with self.mock_config_info({'key': 'abcdefg'}), \
+                self.mock_http_conversation('test_403'):
+                    with self.mock_http_conversation('test_languages'):
+                        self.initialize_bot()
+                    self.assert_bot_response(
+                        message = {'content': '"hello" german english', 'sender_full_name': 'tester'},
+                        response = {'content': 'Translate Error. Invalid API Key..'},
+                        expected_method = 'send_reply'
+                    )
+
+    def test_help(self):
```

Comment:
Re the name of this test, this is really just testing getting help text in response to an empty string - there are many cases which also return the help text. These tests would be great to add, whether in this test function (so `test_help` remains descriptive), or in individual tests (in which case renaming this test may be appropriate).